File: SuperLauncher/ui/pages/account_page.py
from PyQt6.QtWidgets import QWidget, QVBoxLayout, QHBoxLayout, QLabel, QPushButton, QLineEdit, QGroupBox
from PyQt6.QtCore import Qt
from PyQt6.QtGui import QCursor

class AccountPage(QWidget):
    def __init__(self, main):
        super().__init__()
        self.main = main
        layout = QVBoxLayout(self)
        layout.setContentsMargins(20, 20, 20, 20)
        layout.setSpacing(15)

        title = QLabel("\U0001F464 Мой Аккаунт")
        title.setStyleSheet("font-size: 26px; font-weight: bold; color: white;")
        layout.addWidget(title)

        info = QGroupBox("Информация")
        info.setStyleSheet("QGroupBox { color: #4facfe; font-weight: bold; border: 1px solid #4facfe; border-radius: 10px; margin-top: 10px; padding-top: 15px; } QGroupBox::title { subcontrol-origin: margin; left: 10px; padding: 0 8px; }")
        info_layout = QVBoxLayout(info)
        self.name_lbl = QLabel("Имя: Гость")
        self.name_lbl.setStyleSheet("color: white; font-size: 14px; padding: 5px;")
        self.status_lbl = QLabel("Статус: Бесплатная версия")
        self.status_lbl.setStyleSheet("color: white; font-size: 14px; padding: 5px;")
        info_layout.addWidget(self.name_lbl)
        info_layout.addWidget(self.status_lbl)
        layout.addWidget(info)

        license_group = QGroupBox("Лицензия")
        license_group.setStyleSheet("QGroupBox { color: #4facfe; font-weight: bold; border: 1px solid #4facfe; border-radius: 10px; margin-top: 10px; padding-top: 15px; } QGroupBox::title { subcontrol-origin: margin; left: 10px; padding: 0 8px; }")
        lic_layout = QVBoxLayout(license_group)
        self.key_input = QLineEdit()
        self.key_input.setPlaceholderText("Введите ключ лицензии...")
        lic_layout.addWidget(self.key_input)
        activate_btn = QPushButton("Активировать")
        activate_btn.clicked.connect(self._activate)
        activate_btn.setCursor(Qt.CursorShape.PointingHandCursor)
        lic_layout.addWidget(activate_btn)
        layout.addWidget(license_group)

        login_btn = QPushButton("Войти / Зарегистрироваться")
        login_btn.clicked.connect(self._login)
        login_btn.setCursor(Qt.CursorShape.PointingHandCursor)
        layout.addWidget(login_btn)

        self.logout_btn = QPushButton("Выйти")
        self.logout_btn.clicked.connect(self._logout)
        self.logout_btn.setCursor(Qt.CursorShape.PointingHandCursor)
        self.logout_btn.setStyleSheet("QPushButton { background: #f44336; }")
        self.logout_btn.hide()
        layout.addWidget(self.logout_btn)
        layout.addStretch()

    def _activate(self):
        key = self.key_input.text().strip()
        if not key:
            QMessageBox.warning(self, "Ошибка", "Введите ключ")
            return
        from PyQt6.QtWidgets import QMessageBox
        if not self.main.account.current_user:
            QMessageBox.warning(self, "Ошибка", "Требуется вход")
            return
        ok, msg = self.main.account.activate_license(key, self.main.account.current_user["user_id"])
        if ok:
            QMessageBox.information(self, "Успех", msg)
        else:
            QMessageBox.critical(self, "Ошибка", msg)

    def _login(self):
        from ..dialogs.login_dialog import LoginDialog
        dlg = LoginDialog(self.main.account, self)
        if dlg.exec():
            if dlg.user_data:
                self.name_lbl.setText(f"Имя: {dlg.user_data['username']}")
                self.status_lbl.setText(f"Статус: {dlg.user_data.get('license_tier', 'free').upper()}")
                self.logout_btn.show()

    def _logout(self):
        self.main.account.logout()
        self.name_lbl.setText("Имя: Гость")
        self.status_lbl.setText("Статус: Бесплатная версия")
        self.logout_btn.hide()

File: SuperLauncherCpp/src/ui/pages/AccountPage.cpp
#include "AccountPage.h"
#include "src/ui/widgets/AnimatedButton.h"
#include "src/ui/widgets/GlassFrame.h"
#include <QStackedWidget>

AccountPage::AccountPage(QWidget *parent)
    : QWidget(parent)
{
    setupUi();
    showLoginForm();
}

void AccountPage::setupUi()
{
    auto *mainLayout = new QVBoxLayout(this);
    mainLayout->setContentsMargins(40, 40, 40, 40);

    auto *title = new QLabel("Аккаунт", this);
    title->setStyleSheet("color: white; font-size: 24px; font-weight: bold;");
    mainLayout->addWidget(title);

    m_stack = new QStackedWidget(this);
    mainLayout->addWidget(m_stack);

    // --- Login Form ---
    m_loginForm = new QWidget(this);
    auto *loginLayout = new QVBoxLayout(m_loginForm);
    loginLayout->setSpacing(12);

    m_loginForm->setStyleSheet("QLineEdit { background: rgba(255,255,255,0.08); "
                                "color: white; border: 1px solid rgba(255,255,255,0.15); "
                                "border-radius: 6px; padding: 10px; font-size: 14px; }"
                                "QLineEdit:focus { border-color: #7C4DFF; }");

    auto *lk = new QLabel("Вход", m_loginForm);
    lk->setStyleSheet("color: #AAAAAA; font-size: 14px;");

    m_loginUsername = new QLineEdit(m_loginForm);
    m_loginUsername->setPlaceholderText("Логин или Email");

    m_loginPassword = new QLineEdit(m_loginForm);
    m_loginPassword->setPlaceholderText("Пароль");
    m_loginPassword->setEchoMode(QLineEdit::Password);

    m_showPassword = new QCheckBox("Показать пароль", m_loginForm);
    m_showPassword->setStyleSheet("color: #888888;");
    connect(m_showPassword, &QCheckBox::toggled, this, [this](bool checked) {
        m_loginPassword->setEchoMode(checked ? QLineEdit::Normal : QLineEdit::Password);
    });

    auto *loginBtn = new AnimatedButton("Войти", m_loginForm);
    loginBtn->setStyleSheet(
        "QPushButton { background: #7C4DFF; color: white; font-size: 14px; "
        "border-radius: 6px; } QPushButton:hover { background: #9C6DFF; }");
    connect(loginBtn, &QPushButton::clicked, this, [this]() {
        emit loginRequested(m_loginUsername->text(), m_loginPassword->text());
    });

    loginLayout->addWidget(lk);
    loginLayout->addWidget(m_loginUsername);
    loginLayout->addWidget(m_loginPassword);
    loginLayout->addWidget(m_showPassword);
    loginLayout->addWidget(loginBtn);
    loginLayout->addStretch();

    // Registration fields
    auto *regTitle = new QLabel("Регистрация", m_loginForm);
    regTitle->setStyleSheet("color: #AAAAAA; font-size: 14px; margin-top: 20px;");

    m_regUsername = new QLineEdit(m_loginForm);
    m_regUsername->setPlaceholderText("Логин");

    m_regEmail = new QLineEdit(m_loginForm);
    m_regEmail->setPlaceholderText("Email");

    m_regPassword = new QLineEdit(m_loginForm);
    m_regPassword->setPlaceholderText("Пароль");
    m_regPassword->setEchoMode(QLineEdit::Password);

    auto *regBtn = new AnimatedButton("Зарегистрироваться", m_loginForm);
    regBtn->setStyleSheet(
        "QPushButton { background: #448AFF; color: white; font-size: 14px; "
        "border-radius: 6px; } QPushButton:hover { background: #64AAFF; }");
    connect(regBtn, &QPushButton::clicked, this, [this]() {
        emit registerRequested(m_regUsername->text(), m_regEmail->text(),
                                m_regPassword->text());
    });

    loginLayout->addWidget(regTitle);
    loginLayout->addWidget(m_regUsername);
    loginLayout->addWidget(m_regEmail);
    loginLayout->addWidget(m_regPassword);
    loginLayout->addWidget(regBtn);
    m_stack->addWidget(m_loginForm);

    // --- Profile ---
    m_profileWidget = new QWidget(this);
    auto *profLayout = new QVBoxLayout(m_profileWidget);
    profLayout->setSpacing(12);

    m_usernameLabel = new QLabel(m_profileWidget);
    m_usernameLabel->setStyleSheet("color: white; font-size: 20px; font-weight: bold;");

    m_emailLabel = new QLabel(m_profileWidget);
    m_emailLabel->setStyleSheet("color: #AAAAAA; font-size: 13px;");

    m_licenseLabel = new QLabel(m_profileWidget);
    m_licenseLabel->setStyleSheet("color: #FFD700; font-size: 13px;");

    m_levelLabel = new QLabel(m_profileWidget);
    m_levelLabel->setStyleSheet("color: #7C4DFF; font-size: 13px;");

    m_xpLabel = new QLabel(m_profileWidget);
    m_xpLabel->setStyleSheet("color: #888888; font-size: 12px;");

    // License activation
    auto *licTitle = new QLabel("Активация лицензии", m_profileWidget);
    licTitle->setStyleSheet("color: #AAAAAA; font-size: 14px; margin-top: 20px;");

    m_licenseKey = new QLineEdit(m_profileWidget);
    m_licenseKey->setPlaceholderText("XXXXXX-XXXXXX-XXXXXX-XXXXXX");
    m_licenseKey->setStyleSheet("background: rgba(255,255,255,0.08); color: white; "
                                 "border: 1px solid rgba(255,255,255,0.15); "
                                 "border-radius: 6px; padding: 10px;");

    auto *activateBtn = new AnimatedButton("Активировать", m_profileWidget);
    activateBtn->setStyleSheet(
        "QPushButton { background: #FFD700; color: #333; font-size: 14px; "
        "border-radius: 6px; } QPushButton:hover { background: #FFE44D; }");
    connect(activateBtn, &QPushButton::clicked, this, [this]() {
        emit activateLicense(m_licenseKey->text());
    });

    auto *logoutBtn = new AnimatedButton("Выйти", m_profileWidget);
    logoutBtn->setStyleSheet(
        "QPushButton { background: #f44336; color: white; font-size: 14px; "
        "border-radius: 6px; } QPushButton:hover { background: #da190b; }");
    connect(logoutBtn, &QPushButton::clicked, this, &AccountPage::logoutRequested);

    profLayout->addWidget(m_usernameLabel);
    profLayout->addWidget(m_emailLabel);
    profLayout->addWidget(m_licenseLabel);
    profLayout->addWidget(m_levelLabel);
    profLayout->addWidget(m_xpLabel);
    profLayout->addWidget(licTitle);
    profLayout->addWidget(m_licenseKey);
    profLayout->addWidget(activateBtn);
    profLayout->addWidget(logoutBtn);
    profLayout->addStretch();
    m_stack->addWidget(m_profileWidget);
}

void AccountPage::showLoginForm()
{
    m_stack->setCurrentWidget(m_loginForm);
}

void AccountPage::showProfile()
{
    m_stack->setCurrentWidget(m_profileWidget);
}

void AccountPage::setLoggedIn(bool loggedIn, const QString &username,
                               const QString &tier)
{
    if (loggedIn) {
        m_usernameLabel->setText(username);
        m_licenseLabel->setText("Лицензия: " + (tier.isEmpty() ? "free" : tier));
        showProfile();
    } else {
        showLoginForm();
    }
}

void AccountPage::setUserInfo(const QString &username, const QString &email,
                               const QString &license, int xp, int level)
{
    m_usernameLabel->setText(username);
    m_emailLabel->setText(email);
    m_licenseLabel->setText("Лицензия: " + license);
    m_xpLabel->setText("Опыт: " + QString::number(xp));
    m_levelLabel->setText("Уровень: " + QString::number(level));
}
